src/test/java/org/broadinstitute/hellbender/utils/config/ConfigUnitTest.java

```diff
@@ -0,0 +1,650 @@
+
+        // List properties for inspection:
+        listAndStoreConfigToStdOut(basicTestConfigWithClassPathOverridesAndVariableFile);
+
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.booleanDefFalse(), true);
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.booleanDefTrue(), false);
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.intDef207(), 702);
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.listOfStringTest(), new ArrayList<>(Arrays.asList("string4", "string3", "string2", "string1")));
+        Assert.assertEquals(basicTestConfigWithClassPathOverridesAndVariableFile.customBoolean().booleanValue(), true);
+
+        // Reset the config factory:
+        ConfigFactory.clearProperty(BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME);
+        Assert.assertEquals(ConfigFactory.getProperty(BasicTestConfigWithClassPathOverridesAndVariableFile.CONFIG_FILE_VARIABLE_FILE_NAME), null);
+    }
+
+    @Test(dataProvider = "createSourcesFilenamesAndSystemPropertiesFlags")
+    void testConfigUtilsCheckFileNamePropertyExistenceAndSetConfigFactoryProperties(final List<String> sourcesFilenames,
+                                                                                    final List<Boolean> addToSystemProperties) {
+        Assert.assertEquals(sourcesFilenames.size(), addToSystemProperties.size());
+
+        // Ensure that we don't have any of these in our system properties or environment:
+        for( final String fileNameVar : sourcesFilenames ) {
+            Assert.assertEquals(System.getenv(fileNameVar), null);
+            Assert.assertEquals(System.getProperty(fileNameVar), null);
+        }
+
+        // Add in the relevant values to our system properties:
+        for ( int i = 0 ; i < sourcesFilenames.size() ; ++i ) {
+            if ( addToSystemProperties.get(i) ) {
+                System.setProperty(sourcesFilenames.get(i), sourcesFilenames.get(i));
+            }
+        }
+
+        // Set up our config factory values:
+        ConfigUtils.checkFileNamePropertyExistenceAndSetConfigFactoryProperties(sourcesFilenames);
+
+        // Now we check them:
+        for ( int i = 0 ; i < sourcesFilenames.size() ; ++i ) {
+
+            // If we added them to the properties, we don't need to set the names in
+            // the ConfigFactory.  Therefore they should be null:
+            if ( addToSystemProperties.get(i) ) {
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), null);
+            }
+            // If the filename didn't exist in the system properties, then we did need to add them to the
+            // ConfigFactory properties:
+            else {
+                Assert.assertEquals(ConfigFactory.getProperty(sourcesFilenames.get(i)), ConfigUtils.NO_PATH_VARIABLE_VALUE);
```

Comment:
Can you explain the importance of this distinction between `null` and `/dev/null` here?

Reply:
The `Owner::ConfigFactory` doesn't allow the user to set `null` as a property.  It results in a `NullPointerException`.  So since we only officially support Linux-style OSs, I thought `/dev/null` was a good sentinel value for having no path variable.